flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogDatabase.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import java.util.Map;
+
+/**
+ * Represents a database object in a catalog.
+ */
+public interface CatalogDatabase {
```

Comment:
As can be seen throughout the design and code here, every meta class has properties. As to database, it can have properties like created_at, created_by, last_ddl_time, etc. Maybe not every catalog need it, but I think in general we need to support those on the API level.

Reply:
Do we know of any catalogs that would need that concept? I don't know I don't see any benefits in having those classes right now. Those properties you mentioned are system properties, which should be auto generated, are we going to do that for all catalogs?

I really feel that we do not have any use cases for those classes. Could we do the same as with "moving tables between databases" and introduce it once we know what do we need them for? It's always easier to introduce new API than remove old one.

Reply:
In the Catalog, we can have multiple `CatalogDatabases`. There are multiple `CatalogTables` and `CatalogViews` in each `CatalogDatabase`. I think the meaning of the `CatalogDatabase` is the management of the CommonTable. The management relationship here is a bit special. For example, the `CatalogDatabase` and the CommonTable are the members of XXXReadableCatalog, and members are flat managed. They are dependent on `ObjectPath`. For example, `ObjectPath` contains the databaseName attribute. This management method is similar to the way that spark integrates hive, and it is flat management. Please see: https://github.com/apache/spark/blob/bbbe54aa79c0f6b66e3f3ac34515cc096beb5730/sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala#L237

Feel free to correct me if there is anything I misunderstood. :-)

Reply:
I totally get the multi stage referencing and paths and I am totally for them, but we don't need the class `CatalogDatabes` for that. I just don't see a use cases for having an independent `CatalogDatabase` representation. 

When and what for would we need to do `Catalog.getDatabase()`?

Reply:
From the points of view, the user may want to see all databases in the catalog by `listDatabases`. and if user wants to see the detail for database info, such as properties, will call `getDatabase(..)`.

Reply:
> Hi @sunjincheng121 , @xuefuz , @bowenli86
> 
> Regarding to the `listTables()`, I'm fine with current approach that `listTables` returns tables and views if you all prefer it. I don't have a strong opinion on this.
> 
> Regarding moving to the mailing list, my initial intention is involving more people into the discussion when we have diverging opinions on API, but it seems that we have a consensus on this now. I agree with @sunjincheng121 that if we think the API change is minor, I think it's fine to keep the discussion in PR. But I would suggest to update the FLIP according to the latest change after the pull request is merged.
> 
> So, I'm +1 to merge this after Dawid's comments addressed.

Yes, I will update the FLIP to reflect what we have agreed upon. Thanks.